src/components/Personalization/helper/click.js

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { uuid } from "../../../utils";
+import { awaitSelector } from "../../../utils/dom";
+import ALLOY_KEY from "../../../constants/clickTrackingAttribute";
+import { selectNodesWithEq, setAttribute } from "./dom";
+
+export default (settings, store) => {
+  const key = uuid();
+  const { selector, meta } = settings;
+
+  return awaitSelector(selector, selectNodesWithEq).then(elements => {
+    store(key, meta);
+
+    elements.forEach(element => {
+      setAttribute(element, ALLOY_KEY, key);
+    });
```

Comment:
Do we ever have elements that are related to multiple `settings` objects? If so, the `data-alloy-key` UUID that was stored on the element from the first `settings` object will get overwritten by the UUID created for the second `settings` object, which I assume would be a bad thing.

Reply:
@Aaronius the `settings` will contain a `selector` and we try to decorate ALL the elements matched by that `selector`. I am not sure what you mean by "multiple settings object".

Reply:
Sorry I wasn't clear. Let's say there's a rule with an action with a `moduleType` of `click` and the value of its `settings.selector` is `.foo`. Then let's say there's another rule with an action with a `moduleType` of `click` and the value of its `settings.selector` is `.bar`. Then let's say there's an element on the page with `class="foo bar"`. In this case, if I'm following the code right (which I very well may not), when the first rule executes, a UUID will be generated and applied to the element's `data-alloy-key` attribute and meta will be stored using that UUID. Then, when the second rule executes, a new UUID will be generated for the exact same element. That UUID would be used to override the `data-alloy-key` attribute that had previously been applied to that element. I'm guessing that would foul up the tracking part of the process.

Reply:
To solve this you could put multiple values in the key data element. (maybe separated by space).